pkg/controllers/nodeclaim/inplaceupdate/controller.go

```diff
@@ -44,13 +44,17 @@ import (
 type Controller struct {
 	kubeClient       client.Client
 	instanceProvider *instance.Provider
+
+	// Will be used to calculate the goal state
+	opts *options.Options
```

Comment:
Another instance of "not sure if options is overkill", see the comments in `operator.go` for similar discussions.